In [1]:
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

MessageError: Error: credential propagation was unsuccessful

In [ ]:

# Importing packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [ ]:
# Reading hvcan data from csv file

hvcan = pd.read_csv("/content/drive/MyDrive/Honda Capstone Project- MOVE/hvcan_1-22.csv")

hvcan.head()

uuid                         segmentid partitionDate  \
0  1641042438772103077  40119a362475ed3d4de98b555786d973    2022-01-01   
1  1641042438869203077  40119a362475ed3d4de98b555786d973    2022-01-01   
2  1641042438969303077  40119a362475ed3d4de98b555786d973    2022-01-01   
3  1641042439069403077  40119a362475ed3d4de98b555786d973    2022-01-01   
4  1641042439169503077  40119a362475ed3d4de98b555786d973    2022-01-01   

   device      timestamp  filedate  filetimeutc      updateTimeMs  sequenceNo  \
0    3077  1641042438772  20220101       130718  1668774677451156           1   
1    3077  1641042438869  20220101       130718  1668774677451292           2   
2    3077  1641042438969  20220101       130718  1668774677451301           3   
3    3077  1641042439069  20220101       130718  1668774677451312           4   
4    3077  1641042439169  20220101       130718  1668774677451316           5   

                                vin              tripId  localTimeMs  \
0  663fd3c5144fd10bd5ca6611a9a5b92d  307720220101130718     47238772   
1  663fd3c5144fd10bd5ca6611a9a5b92d  307720220101130718     47238869   
2  663fd3c5144fd10bd5ca6611a9a5b92d  307720220101130718     47238969   
3  663fd3c5144fd10bd5ca6611a9a5b92d  307720220101130718     47239069   
4  663fd3c5144fd10bd5ca6611a9a5b92d  307720220101130718     47239169   

   engineTorque  engineCoolantTemp  engineAtmosphericPressure  \
0        6553.4                -41                       91.4   
1        6553.4                -41                       91.4   
2        6553.4                -41                       91.4   
3        6553.4                -41                       91.4   
4        6553.4                -41                       91.4   

   fileVersionNumber  steeringSpeed  engineSpeed  steeringTorque  epsLoad  \
0                  2              0            0             0.0      0.0   
1                  2              0            0             0.0      0.0   
2                  2              0            0             0.0      0.0   
3                  2              0            0             0.0      0.0   
4                  2              0            0             0.0      0.0   

   brakePressure  torqueControlModeFast  torqueControlModeSlow  \
0            0.0                      0                      0   
1            0.0                      0                      0   
2            0.0                      0                      0   
3            0.0                      0                      0   
4            0.0                      0                      0   

   parkBrakeStatus  warnStatusAcc  dimmerCancel  passenger2Belt  \
0            False          False         False           False   
1            False          False         False           False   
2            False          False         False           False   
3            False          False         False           False   
4            False          False         False           False   

   passengerOccupied  rearLeftBelt  rearCenterBelt  rearRightBelt  accWarning  \
0              False         False           False          False       False   
1              False         False           False          False       False   
2              False         False           False          False       False   
3              False         False           False          False       False   
4              False         False           False          False       False   

   accLsfEngage  accLsfSubseg  accControlMode  accCmbsSetting  fuelUsed  \
0         False             0           False               0       0.0   
1         False             0           False               0       0.0   
2         False             0           False               0       0.0   
3         False             0           False               0       0.0   
4         False             0           False               0       0.0   

   lkasReady  lkasShowLane  cabinTemp  warnStatusParking  war

In [ ]:
# Checking dimensions of hvcan data

hvcan.shape

(1372877, 80)

In [ ]:
# Checking data values in hvvan data

hvcan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1372877 entries, 0 to 1372876
Data columns (total 80 columns):
 #   Column                     Non-Null Count    Dtype  
---  ------                     --------------    -----  
 0   uuid                       1372877 non-null  object 
 1   segmentid                  1372877 non-null  object 
 2   partitionDate              1372877 non-null  object 
 3   device                     1372877 non-null  int64  
 4   timestamp                  1372877 non-null  int64  
 5   filedate                   1372877 non-null  int64  
 6   filetimeutc                1372877 non-null  int64  
 7   updateTimeMs               1372877 non-null  int64  
 8   sequenceNo                 1372877 non-null  int64  
 9   vin                        1372877 non-null  object 
 10  tripId                     1372877 non-null  int64  
 11  localTimeMs                1372877 non-null  int64  
 12  engineTorque               1372877 non-null  float64
 13  engineCoolan

In [ ]:
# Checking numerical attributes of hvcan data

#hvcan.describe()

In [ ]:
# checking numerical data type of hvcan

#num_cols = hvcan.select_dtypes(exclude = ['object', 'boolean'])

#print(len(num_cols.columns))
#num_cols.columns

In [ ]:
# Checking categorical data type of hvcan

#cat_cols = hvcan.select_dtypes(include = 'object')

#print(len(cat_cols.columns))
#cat_cols.columns

In [ ]:
# Checking boolean data type of hvcan

#bool_cols = hvcan.select_dtypes(include= 'boolean', exclude= ['object', 'integer', 'float'])

#print(len(bool_cols.columns))
#bool_cols.columns

In [ ]:
# Checking dataframe hvcan

#hvcan.head()

In [ ]:
# Checking numerical values of hvcan data

#hvcan.describe()

In [ ]:
# Dropping columns with zero values and id columns

hvcan = hvcan.drop(['brakePressure',
                    'torqueControlModeFast',
                    'torqueControlModeSlow',
                    'accCmbsSetting',
                    'engineIntakeAirTemp',
                    'engineEcompNe',
                    'tcuTrqConfCl2',
                    'uuid',
                    'segmentid',
                    'sequenceNo',
                    'vin',
                    'tripId',
                    'timestamp',
                    'updateTimeMs',
                    'localTimeMs'], axis = 1)

In [ ]:
# Checking hvcan data

#hvcan.head()

In [ ]:
# Checking hvcan dataframe
#hvcan.shape

# Univariate analysis of data

### 1. PartitionDate Column

In [ ]:
#plt.figure(figsize= (15, 8))

#hvcan['partitionDate'].value_counts(dropna=False).plot(kind ='bar')
#plt.show()

In [ ]:
# Converting partitionDate to datetime

hvcan['partitionDate'] = pd.to_datetime(hvcan['partitionDate'])

### 2. Column device

In [ ]:
#hvcan['device'].value_counts(dropna=False)

In [ ]:
#plt.figure(figsize = (15, 6))

#hvcan['device'].value_counts(dropna=False).plot(kind='bar')

#plt.show()

### Inferences-
- Selecting devices 3053, 3045, 3040, 3009, 3027, 3046, 3077, 3002, 3071 for study based on their occurance in the hvcan dataset

### 3. Column engineTorque

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['engineTorque'])
#plt.show()

In [ ]:
# Dropping outlier values from column engineTorque

#quant_1 = hvcan['engineTorque'].quantile(0.99)
#quant_2 = hvcan['engineTorque'].quantile(0.01)

#hvcan = hvcan[(hvcan['engineTorque'] < quant_1) & (hvcan['engineTorque'] > quant_2)]

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['engineTorque'])
#plt.show()

## 4. Column engineCoolantTemp

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.histplot(hvcan['engineCoolantTemp'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['engineCoolantTemp'])
#plt.show()

### Inferences:

- Histogram shows majority value distribution towards zero and box plot shos value distribution between 40 to 80

## 5. Column- engineAtmosphericPressure

In [ ]:
#plt.figure(figsize=(15, 6))
#hvcan['engineAtmosphericPressure'].value_counts().plot(kind='bar')
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['engineAtmosphericPressure'])
#plt.show()

In [ ]:
#hvcan['engineAtmosphericPressure'].describe()

### Inferences:

- Histogram shows majority value distribution towards 91.4 and 0
- Box plot shos value distribution between 100 to 700+

## 6. Column - fileVersionNumber

In [ ]:
#hvcan['fileVersionNumber'].value_counts()

In [ ]:
#plt.figure(figsize=(10, 5))
#hvcan['fileVersionNumber'].value_counts().plot(kind='bar')
#plt.show()

In [ ]:
# Checking value distribution if fileVersionNumber column is useful for analysis

#100*(hvcan['fileVersionNumber'].value_counts()/hvcan.shape[0])

In [ ]:
# From value distribution dropping column fileVersionNumber as data is skewed towards version 2

hvcan = hvcan.drop('fileVersionNumber', axis=1)

#hvcan.head()

## 7. Column steeringSpeed

In [ ]:
#plt.figure(figsize=(15,5))
#sns.histplot(hvcan['steeringSpeed'])
#plt.show()

In [ ]:

#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['steeringSpeed'])
#plt.show()

## 8. Column engineSpeed

In [ ]:
#plt.figure(figsize=(15,5))
#sns.histplot(hvcan['engineSpeed'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['engineSpeed'])
#plt.show()

In [ ]:
#quant_1 = hvcan['engineSpeed'].quantile(0.99)

#hvcan = hvcan[(hvcan['engineSpeed'] < quant_1)]

In [ ]:
#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['engineSpeed'])
#plt.show()

## 9. Column steeringTorque

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['steeringTorque'])
#plt.show()

## 10. Column epsLoad

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['epsLoad'])
#plt.show()

In [ ]:
#quant_1 = hvcan['epsLoad'].quantile(0.99)

#hvcan = hvcan[(hvcan['epsLoad'] < quant_1)]

In [ ]:
#plt.figure(figsize=(15, 6))
#sns.boxplot(hvcan['epsLoad'])
#plt.show()

In [ ]:
#hvcan.head()

## 11. Column parkBrakeStatus

In [ ]:
#hvcan['parkBrakeStatus'].value_counts()

In [ ]:
#hvcan['parkBrakeStatus'].value_counts().plot(kind='bar')

In [ ]:
#100*(hvcan['parkBrakeStatus'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping 'parkBrakeStatus' from hvcan

hvcan = hvcan.drop('parkBrakeStatus', axis= 1)

#hvcan.head()

## 12. Column warnStatusAcc

In [ ]:
#hvcan['warnStatusAcc'].value_counts()

In [ ]:
# Dropping 'warnStatusAcc' from hvcan

hvcan = hvcan.drop('warnStatusAcc', axis = 1)

#hvcan.head()

## 13. Column dimmerCancel

In [ ]:
#hvcan['dimmerCancel'].value_counts()

In [ ]:
#hvcan['dimmerCancel'].value_counts().plot(kind='bar')

In [ ]:
#100*(hvcan['dimmerCancel'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping 'dimmerCancel' from hvcan

hvcan = hvcan.drop('dimmerCancel', axis= 1)
#hvcan.head()

## 14. Column passenger2Belt

In [ ]:
#hvcan['passenger2Belt'].value_counts()

In [ ]:
hvcan = hvcan.drop('passenger2Belt', axis=1)

#hvcan.head()

## 15. Column passengerOccupied

In [ ]:
#hvcan['passengerOccupied'].value_counts()

In [ ]:
hvcan = hvcan.drop('passengerOccupied', axis=1)

#hvcan.head()

## 16. Column rearLeftBelt

In [ ]:
#hvcan['rearLeftBelt'].value_counts()

In [ ]:
#100*(hvcan['rearLeftBelt'].value_counts()/hvcan.shape[0])

In [ ]:
hvcan = hvcan.drop('rearLeftBelt', axis= 1)

#hvcan.head()

## 17. Column rearCenterBelt

In [ ]:
#hvcan['rearCenterBelt'].value_counts()

In [ ]:
#100*(hvcan['rearCenterBelt'].value_counts()/hvcan.shape[0])


In [ ]:
hvcan = hvcan.drop('rearCenterBelt', axis=1)
#hvcan.head()

## 18. Column rearRightBelt

In [ ]:
#hvcan['rearRightBelt'].value_counts()

In [ ]:
#100*(hvcan['rearRightBelt'].value_counts()/hvcan.shape[0])


In [ ]:
hvcan = hvcan.drop('rearRightBelt', axis=1)

#hvcan.head()

## 19. Column accWarning

In [ ]:
# Checking value distribution
#hvcan['accWarning'].value_counts()

In [ ]:
# Checking percentage value distribution
#100*(hvcan['accWarning'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column accWarning as sublevels are imbalances
hvcan = hvcan.drop('accWarning', axis=1)
#hvcan.head()

## 20. Column accLsfEngage

In [ ]:
#hvcan['accLsfEngage'].value_counts()

In [ ]:
#100*(hvcan['accLsfEngage'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column accLsfEngage as sublevels are imbalances

hvcan = hvcan.drop('accLsfEngage', axis=1)

#hvcan.head()

## 21. Column accLsfSubseg

In [ ]:
#hvcan['accLsfSubseg'].value_counts()

In [ ]:
#100*(hvcan['accLsfSubseg'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column accLsfSubseg as sublevels are imbalances

#hvcan = hvcan.drop('accLsfSubseg', axis=1)

#hvcan.head()

## 22. Column accControlMode

In [ ]:
#hvcan['accControlMode'].value_counts()

In [ ]:
#100*(hvcan['accControlMode'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column accControlMode as sublevels are imbalances

hvcan = hvcan.drop('accControlMode', axis=1)

#hvcan.head()

## 23. Column fuelUsed

In [ ]:
#plt.figure(figsize=(15,6))
#sns.histplot(hvcan['fuelUsed'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['fuelUsed'])
#plt.show()

In [ ]:
#quant_1 = hvcan['fuelUsed'].quantile(0.97)

#hvcan = hvcan[(hvcan['fuelUsed'] < quant_1)]

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['fuelUsed'])
#plt.show()

## 24. Column lkasReady

In [ ]:
#hvcan['lkasReady'].value_counts()

In [ ]:
#100*(hvcan['lkasReady'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column lkasReady as sublevels are imbalances

hvcan = hvcan.drop('lkasReady', axis=1)

#hvcan.head()

## 25. Column -lkasShowLane

In [ ]:
#hvcan['lkasShowLane'].value_counts()

In [ ]:
#100*(hvcan['lkasShowLane'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping column lkasShowLane as sublevels are imbalances

hvcan = hvcan.drop('lkasShowLane', axis=1)

#hvcan.head()

## 26. Column cabinTemp

In [ ]:
#plt.figure(figsize=(15,6))
#sns.histplot(hvcan['cabinTemp'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['cabinTemp'])
#plt.show()

In [ ]:
#hvcan.head()

## 27. Column warnStatusParking

In [ ]:
#hvcan['warnStatusParking'].value_counts()

In [ ]:
#100*(hvcan['warnStatusParking'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping 'warnStatusParking' column

hvcan = hvcan.drop('warnStatusParking', axis=1)

#hvcan.head()

## 28. Column - warnStatusCmbs

In [ ]:
#hvcan['warnStatusCmbs'].value_counts()

In [ ]:
# Dropping warnStatusCmbs from hvcan

hvcan = hvcan.drop('warnStatusCmbs', axis= 1)

#hvcan.head()

## 29. Column - warnStatusEbd

In [ ]:
#hvcan['warnStatusEbd'].value_counts()

In [ ]:
# Dropping warnStatusEbd from hvcan data

hvcan = hvcan.drop('warnStatusEbd', axis=1)

#hvcan.head()

## 30. Column - compCurrent

In [ ]:
#plt.figure(figsize=(15,6))
#sns.histplot(hvcan['compCurrent'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['compCurrent'])
#plt.show()

In [ ]:
quant_1 = hvcan['compCurrent'].quantile(0.99)
hvcan = hvcan[hvcan['compCurrent'] < quant_1]

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['compCurrent'])
#plt.show()

## 31. Column - fogLightRelay

In [ ]:
#hvcan['fogLightRelay'].value_counts()

In [ ]:
# Dropping fogLightRelay from hvcan

hvcan = hvcan.drop('fogLightRelay', axis=1)

#hvcan.head()

## 32. Column -airbagStatus

In [ ]:
#hvcan['airbagStatus'].value_counts()

In [ ]:
# Dropping airbagStatus from hvcan

hvcan = hvcan.drop('airbagStatus', axis=1)

#hvcan.head()

## 33. Column - latAccel

In [ ]:
#plt.figure(figsize=(15,6))
#sns.histplot(hvcan['latAccel'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['latAccel'])
#plt.show()

In [ ]:
#hvcan.head()

## 34. Column - accelPedalPos

In [ ]:
#plt.figure(figsize=(15,6))
#sns.histplot(hvcan['accelPedalPos'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,6))
#sns.boxplot(hvcan['accelPedalPos'])
#plt.show()

## 35. Column - cruiseOverride

In [ ]:
#hvcan['cruiseOverride'].value_counts()

In [ ]:
# Dropping cruiseOverride from hvcan

hvcan = hvcan.drop('cruiseOverride', axis=1)

#hvcan.head()

## 36. Column - cruiseEngaged

In [ ]:
#hvcan['cruiseEngaged'].value_counts()

In [ ]:
#100*(hvcan['cruiseEngaged'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping 'cruiseEngaged' column from hvcan

hvcan = hvcan.drop('cruiseEngaged', axis=1)

#hvcan.head()

## 37. Column - wheelVelLF

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['wheelVelLF'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['wheelVelLF'])
#plt.show()

## 38. Column - wheelVelRF

In [ ]:
#plt.figure(figsize=(15,5))
#sns.histplot(hvcan['wheelVelRF'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['wheelVelRF'])
#plt.show()

## 39. Column- wheelVelLR

In [ ]:
#plt.figure(figsize=(15,5))
#sns.histplot(hvcan['wheelVelLR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['wheelVelLR'])
#plt.show()

## 40. Column - wheelVelRR

In [ ]:
#plt.figure(figsize=(15,5))
#sns.histplot(hvcan['wheelVelRR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15,5))
#sns.boxplot(hvcan['wheelVelRR'])
#plt.show()

In [ ]:
#hvcan.head()

## 41. Column - panicBrkAct


In [ ]:
#hvcan['panicBrkAct'].value_counts()


In [ ]:
# Dropping panicBrkAct from hvcan

hvcan = hvcan.drop('panicBrkAct', axis=1)

#hvcan.head()

## 42. Column - accStatus

In [ ]:
#hvcan['accStatus'].value_counts()

In [ ]:
#plt.figure(figsize=(15,5))
#hvcan['accStatus'].value_counts().plot(kind='bar')
#plt.show()

## 43. Column - odometerMiles

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['odometerMiles'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['odometerMiles'])
#plt.show()

In [ ]:
#hvcan.head()

## 44. Column - pasgSeatbelt

In [ ]:
#hvcan['pasgSeatbelt'].value_counts()

In [ ]:
# Dropping pasgSeatbelt from hvcan

hvcan = hvcan.drop('pasgSeatbelt', axis=1)

#hvcan.head()

## 45. Column - driverSeatbelt

In [ ]:
#hvcan['driverSeatbelt'].value_counts()

In [ ]:
#100*(hvcan['driverSeatbelt'].value_counts()/hvcan.shape[0])

## 46. Column - pasgSeated

In [ ]:
#hvcan['pasgSeated'].value_counts()

In [ ]:
#100*(hvcan['pasgSeated'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping pasgSeatbelt from hvcan

hvcan = hvcan.drop('pasgSeated', axis=1)

#hvcan.head()

## 47. Column - accSetSpeed

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['accSetSpeed'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['accSetSpeed'])
#plt.show()

## 48. Column - cruiseSetSpeed

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['cruiseSetSpeed'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['cruiseSetSpeed'])
#plt.show()

## 49. Column - tirePressureLF

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['tirePressureLF'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['tirePressureLF'])
#plt.show()

## 50. Column - tirePressureRF

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['tirePressureRF'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['tirePressureRF'])
#plt.show()

## 51. Column - tirePressureLR

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['tirePressureLR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['tirePressureLR'])
#plt.show()

## 52. Column - tirePressureRR

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['tirePressureRR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['tirePressureRR'])
#plt.show()

## 53. Column - outsideAirTemp



In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['outsideAirTemp'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['outsideAirTemp'])
#plt.show()

## 54. Column - fogLightsOn

In [ ]:
#hvcan['fogLightsOn'].value_counts()

In [ ]:
#100*(hvcan['fogLightsOn'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping fogLightsOn from hvcan

hvcan = hvcan.drop('fogLightsOn', axis=1)

#hvcan.head()

## 55. Column - stabilityControlActEDR

In [ ]:
#hvcan['stabilityControlActEDR'].value_counts()

In [ ]:
# Dropping stabilityControlActEDR from hvcan

hvcan = hvcan.drop('stabilityControlActEDR', axis=1)

#hvcan.head()

## 56. Column - engAccCancelReasonManual

In [ ]:
#hvcan['engAccCancelReasonManual'].value_counts()

In [ ]:
#100*(hvcan['engAccCancelReasonManual'].value_counts()/hvcan.shape[0])

In [ ]:
# Dropping engAccCancelReasonManual from hvcan

hvcan = hvcan.drop('engAccCancelReasonManual', axis=1)

#hvcan.head()

## 57. Column - tmDisplayedGear

In [ ]:
#hvcan['tmDisplayedGear'].value_counts()

In [ ]:
#plt.figure(figsize=(15,5))
#hvcan['tmDisplayedGear'].value_counts().plot(kind='bar')
#plt.show()

## 58. Column - absWheelSpeedFL

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['absWheelSpeedFL'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['absWheelSpeedFL'])
#plt.show()

## 59. Column - absWheelSpeedFR

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['absWheelSpeedFR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['absWheelSpeedFR'])
#plt.show()

## 60. Column - absWheelSpeedRL

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['absWheelSpeedRL'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['absWheelSpeedRL'])
#plt.show()

## 61. Column - absWheelSpeedRR

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.histplot(hvcan['absWheelSpeedRR'])
#plt.show()

In [ ]:
#plt.figure(figsize=(15, 5))
#sns.boxplot(hvcan['absWheelSpeedRR'])
#plt.show()

## 61. Column - timestamp_hr

In [ ]:
#plt.figure(figsize=(30, 5))
#hvcan['timestamp_hr'].value_counts().plot(kind='bar')
#plt.show()

In [ ]:
hvcan = hvcan.drop('timestamp_hr', axis=1)

#hvcan.head()

## 62. Column - updateTimeMs_hr

In [ ]:
#hvcan['updateTimeMs_hr'].value_counts()

In [ ]:
hvcan = hvcan.drop('updateTimeMs_hr', axis=1)

#hvcan.head()

In [ ]:
hvcan.to_csv('cleaned_hvcan.csv')

In [ ]:
from google.colab import files
files.download('cleaned_hvcan.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>